### DISCUSSION TITLE
How to clone repo to run tests

### DISCUSSION THREAD

**Discussion by lvmasterrj** [vicwomg/pikaraoke#409] [Q&A] [answered]
Hello friends. I've done some changes to the old pikaraoke, to suit my needs, but I was still using the old pikaraoke (vlc + pygame). Now I'm giving a try with the new chrome based pikaraoke.

My trouble is that I would like to clone the repo to make my changes and test them, but I'm not being able to do so.

What I did so far was I forled the repo to a branch of my own. On RPi (with fresh new bookworm) I git cloned it, from my branch and copied the setup.sh and requirements.sh to the root of pikaraoke/pikaraoke and ran the ./setup.sh successfully, but when I started it complained about _No module named 'pikaraoke'_ witch I then installed with `pip install pikaraoke` inside my venv.

When I start the app by using the old `python app.py -l 10` or by copying the ./pikaraoke.sh from scripts to pikaraoke/pikaraoke, I get the following error:

![image](https://github.com/user-attachments/assets/81d940d5-c391-4d89-98d7-4336aa96d55d)

I know that I was supposed to install only the pikaraoke by PyPi, but how can I make it work but letting me do my changes to the code that I could in future do a PR?

Thanks.

**Comment by vicwomg** [comment, marked as answer]
Clone the repo as you normally would, then see this section on how to run
and develop pikaraoke. You would use poetry to launch the new code, not
app.py.
https://github.com/vicwomg/pikaraoke?tab=readme-ov-file#developing-pikaraoke

**Reply by lvmasterrj** [reply]
I was trying to run on cli, without the desktop on. Now it works, but with my RPi 3B+ it's strugling to run smooth. I will try overclocking it.

Thanks, by the way. You're the man!
